src/Dapr.Actors/Runtime/ActorStateChange.cs

```diff
@@ -68,5 +70,39 @@ public ActorStateChange(string stateName, Type type, object value, StateChangeKi
         /// The kind of state change for given actor state name.
         /// </value>
         public StateChangeKind ChangeKind { get; }
+
+        /// <summary>
+        /// Gets the time to live for the state.
+        /// </summary>
+        /// <value>
+        /// The time to live for the state.
+        /// </value>
+        /// <remarks>
+        /// If null, the state will not expire.
+        /// </remarks>
+        public DateTimeOffset? TTLExpireTime { get; }
+
+        /// <summary>
+        /// Gets the time to live in seconds for the state.
+        /// </summary>
+        /// <value>
+        /// The time to live for the state.
+        /// </value>
+        /// <remarks>
+        /// If null, the state will not expire.
+        /// </remarks>
+        public int? TTLInSeconds {
```

Comment:
Since this conversion to seconds is only used in a single place (for the specific circumstances of the runtime API), would it make sense to move this logic there and only expose `TTLExpireTime`?  Is there value in exposing both to users of `ActorStateChange`? (Especially since this property is a constantly changing value, being relative to the current date/time, which users may not expect from a property.)

Reply:
Thanks @philliphoff, I've updated the PR to only have hold the `TTLExpireTime` value which gets converted to int seconds when saving the actor state.